eclipse/src/saros/ui/wizards/ConfigurationWizard.java

```diff
@@ -72,19 +72,18 @@ public void addPages() {
   public boolean performFinish() {
     setConfiguration();
 
-    final XMPPAccount accountToConnect;
+    final XMPPAccount accountToConnect = addOrGetXMPPAccount();
 
-    if (!enterXMPPAccountWizardPage.isExistingAccount()) {
-      accountToConnect = addXMPPAccount();
-    } else accountToConnect = null;
+    assert (accountToConnect != null);
 
-    assert accountStore.getActiveAccount() != null;
+    /* it is possible to finish the wizard multiple times
+     * (also it makes no sense) so ensure the behavior is always the same.
+     */
+
+    accountStore.setDefaultAccount(accountToConnect);
 
     if (preferences.getBoolean(PreferenceConstants.AUTO_CONNECT)) {
-      getShell()
-          .getDisplay()
-          .asyncExec(
-              () -> XMPPConnectionSupport.getInstance().connect(accountToConnect, true, false));
+      getShell().getDisplay().asyncExec(() -> XMPPConnectionSupport.getInstance().connect(false));
```

Comment:
Why not pass the determined `accountToConnect` when it is set as the default account, meaning it will be used anyways?

Reply:
Hmm ? See logic above, always make the account from the ConfigurationWizard the default one.

Please read: AUTO_CONNECT = AUTO_CONNECT_ON_STARTUP 

Normally I should present an option on the last page with buttons Finish&Connect Now and Finish (but I guess this is against the Wizard guidelines). So I should add an button for that.